.github/workflows/prgate.yml

```diff
@@ -0,0 +1,95 @@
+# This workflow runs unit tests and integration tests against the last three
+# python versions on windows and ubuntu.
+
+
+name: PR Gate
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  unit-test:
+    name: Unit Tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+        os: [ubuntu-latest, windows-latest]
+
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -e . 
+    - name: Test with pytest
+      run: |
+        pytest
+  
+  ci:
+    name: CI Tests
+    runs-on: ubuntu-latest
+    env:
+      python-version: "3.11"
+      node-version: "19"
+      package-src: "edk2toolext"
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Set up Python ${{ env.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ env.python-version }}
+      
+      - name: Set up Node ${{ env.node-version}}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.node-version}}
+      
+      - name: Install pip dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r docs/user/requirements.txt
+          pip install -e .
+
+      - name: Run code coverage
+        run: |
+          pip install coverage
+          coverage run -m unittest
+      
+      - name: Upload coverage to codecov
+        uses: codecov/codecov-action@v2
+        with:
+          verbose: false
+
+      - name: Install npm dependencies
+        run: |
+          npm install -g markdownlint-cli
+          npm install -g cspell@5.20.0
```

Comment:
why is this version defined here.  But not for markdownlint-cli? And Flake8 uses a version stored in env.  should probably lock versions for all things.

Reply:
Flake8 is a pip module installed (and strongly versioned) in the requirements file. I've locked the version for markdownlint-cli.